torchaudio/pipelines/_wav2vec2/impl.py

```diff
@@ -11,6 +13,22 @@
 __all__ = []
 
 
+class _Wav2Vec2Model(Module):
+    """Wrapper class for Wav2Vec2Model
+
+    This class allows for normalization at the input (see _params["normalize_waveform"])
+    """
+
+    def __init__(self, model: Wav2Vec2Model, normalize_waveform: Optional[bool]):
+        self.model = model
+        self.normalize_waveform = normalize_waveform
+
+    def forward(self, waveforms: Tensor) -> Tuple[Tensor, Optional[Tensor]]:
+        if self.normalize_waveform:
+            waveforms = F.layer_norm(waveforms, waveforms.shape)
+        return self.model(waveforms)
+
```

Comment:
could you also add an `extract_features` method that applies normalization first and call `self.model.extract_features(...)`?

Reply:
`return self.model(waveforms)` will make a call to `extract_features`. So I don't think we need to rewrite that logic, unless I'm misunderstanding something?

Reply:
we need to expose the same set of public methods from the Wav2Vec2Model class.